compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -418,6 +418,18 @@ pub trait Machine<'mir, 'tcx>: Sized {
         Ok(())
     }
 
+    /// Called on places used for in-place function argument and return value handling.
+    ///
+    /// These places need to be protected to make sure the program cannot tell whether the
+    /// argument/return value was actually copied or passed in-place..
+    fn protect_in_place_function_argument(
```

Comment:
I'm not happy with the name for this machine hook, since it is also used for return places ("output arguments"). Any suggestions?

Also I wonder if I should move the `write_uninit` out of this hook and into the place where it is called...  can't imagine an implementation of this hook that wouldn't want to do `write_uninit`.

Reply:
It seems similarly fragile to have to repeat it at both call sites, so 🤷 I don't have a preference.

Arguments seems fine, it's a bit odd from the high level language perspective, but at the MIR level it all becomes a bit confusing anyway.